packages/tizen_app_manager/tizen/src/tizen_app_manager_plugin.cc

```diff
@@ -4,316 +4,386 @@
+      return flutter::EncodableMap();
+    }
+
+    std::string icon_path = tizen_app_info.GetIconPath();
+    std::string executable_path = tizen_app_info.GetExecutablePath();
+    if (tizen_app_info.GetLastError() != APP_MANAGER_ERROR_NONE) {
+      return flutter::EncodableMap();
+    }
+
+    std::string shared_resource_path = tizen_app_info.GetSharedResourcePath();
+    if (tizen_app_info.GetLastError() != APP_MANAGER_ERROR_NONE) {
+      return flutter::EncodableMap();
+    }
+
+    bool is_no_display = tizen_app_info.GetIsNoDisplay();
+    if (tizen_app_info.GetLastError() != APP_MANAGER_ERROR_NONE) {
+      return flutter::EncodableMap();
+    }
+
+    flutter::EncodableMap metadata;
+    tizen_app_info.GetForEachMetadata(
+        [](const char *key, const char *value, void *user_data) {
+          if (key) {
+            flutter::EncodableMap *metadata =
+                static_cast<flutter::EncodableMap *>(user_data);
+            metadata->insert(std::pair<std::string, std::string>(
+                std::string(key), std::string(value)));
+          }
+          return true;
+        },
+        &metadata);
+    if (tizen_app_info.GetLastError() != APP_MANAGER_ERROR_NONE) {
+      return flutter::EncodableMap();
+    }
+
+    flutter::EncodableMap value;
+    value[flutter::EncodableValue("appId")] = app_id;
+    value[flutter::EncodableValue("packageId")] = package_id;
+    value[flutter::EncodableValue("label")] = label;
+    value[flutter::EncodableValue("type")] = type;
+    if (icon_path.size()) {
+      value[flutter::EncodableValue("iconPath")] = icon_path;
+    }
+    value[flutter::EncodableValue("executablePath")] = executable_path;
+    value[flutter::EncodableValue("sharedResourcePath")] = shared_resource_path;
+    value[flutter::EncodableValue("isNoDisplay")] = is_no_display;
+    value[flutter::EncodableValue("metadata")] = metadata;
+
+    return value;
   }
 
-  if (app_id) {
-    free(app_id);
+  void HandleMethodCall(const FlMethodCall &method_call,
+                        std::unique_ptr<FlMethodResult> result) {
+    const auto &arguments = *method_call.arguments();
+
+    const auto &method_name = method_call.method_name();
+    if (method_name == "getCurrentAppId") {
+      GetCurrentId(std::move(result));
+    } else if (method_name == "getAppInfo") {
+      GetApplicationInfo(arguments, std::move(result));
+    } else if (method_name == "getInstalledApps") {
+      GetInstalledApplicationsInfo(std::move(result));
+    } else if (method_name == "isRunning") {
+      ApplicationIsRunning(arguments, std::move(result));
+    } else {
+      result->NotImplemented();
+    }
   }
-}
 
-void TizenAppManagerPlugin::GetApplicationInfo(
-    const flutter::EncodableValue &arguments, MethodResultPtr result) {
-  std::string id = "";
-  char *err_msg = nullptr;
-  app_info_h app_info = nullptr;
+  void RegisterObserver(std::unique_ptr<FlEventSink> &&events) {
```

Comment:
`void SetupChannels (flutter::PluginRegistrar *registrar)`

Please also refactoring this method.
There were several questions in the method.

-The callback of `launch_event_channel_handler`, `terminal_event_channel_handler` may be performed twice in `std::move (events)` in the callback of Launch.
-`launch_event_channel`, `terminate_event_channel` is not stored
- When registering a callback in the same form as the `BatterystatusStreamHandler` class, how to handle `registered_cnt_`

These are the problems that could not be solved because i could not determine how to proceed ...